idaes/apps/grid_integration/multiperiod/price_taker_model.py

```diff
@@ -0,0 +1,1008 @@
+                if isinstance(blk, OperationModelData):
+                    if blk.config.append_lmp_data:
+                        if not LMP_exists:
+                            raise ValueError(
+                                f"OperationModelData has been defined to automatically "
+                                + f"populate LMP data. However, m.LMP does not exist. "
+                                + f"Please run the append_lmp_data function first or set the "
+                                + f"append_lmp_data attribute to False when configuring "
+                                + f"your OperationModelData object."
+                            )
+                        blk.LMP = self.LMP[p]
+
+    def add_capacity_limits(
+        self,
+        op_blk,
+        design_blk,
+        commodity_var,
+        capacity_min,
+        capacity_max,
+        constraint_type,
+        linearization=False,
+    ):
+        """
+        Adds capacity limit constraints of the form:
+                capacity_min * y(t) <= commodity_var(t) <= capacity_max * y(t)
+                ex: P_min * y(t) <= P(t) <= P_max * y(t) [where P(t) is power at time t]
+
+
+        Args:
+            op_blk:             String of the name of the operation model block, ex: ("fs.op_name")
+            design_blk:         String of the name of the design model block, ex: ("m.design_name")
+            commodity_var:      String of the name of the entity on the model the capacity constraints
+                                will be applied to, ex: ("total_power")
+            capacity_min:       String of the name of the minimum capacity, ex: ("min_power_capacity")
+            capacity_max:       String of the name of the maximum capacity, ex: ("max_power_capacity")
+            constraint_type:    String to choose between linear and nonlinear constraints. Valid
+                                inputs are in: ["linear", "nonlinear"]
+            linearization:      Boolean indicating whether linearization is used when constraint_type is
+                                "nonlinear", True to use linearization, False otherwise
+
+        Assumptions/relationship:
+            capacity min <= capacity max
+            capacity min >= 0
+            capacity max >= 0
+        """
+        op_mode = {
+            t: self.mp_model.period[t].find_component(op_blk + ".op_mode")
+            for t in self.mp_model.period
+        }
+        var_commodity = {
+            t: self.mp_model.period[t].find_component(op_blk + "." + commodity_var)
+            for t in self.mp_model.period
+        }
+        max_capacity = self.find_component(design_blk + "." + capacity_max)
+        min_capacity = self.find_component(design_blk + "." + capacity_min)
+
+        # Importing in the necessary variables
+        if not hasattr(self, "range_time_steps"):
+            self.range_time_steps = RangeSet(len(self.mp_model.set_period))
```

Comment:
Can you comment on why this might not exist at this point? It seems to me that this is something that should exist before you get here.